lib/fluent/plugin/out_forward.rb

```diff
@@ -214,22 +214,23 @@ def write(chunk)
       select_a_healthy_node{|node| node.send_data(tag, chunk) }
     end
 
-    ACKWaitingSockInfo = Struct.new(:sock, :chunk_id, :node, :time, :timeout) do
+    ACKWaitingSockInfo = Struct.new(:sock, :chunk_id, :chunk_id_base64, :node, :time, :timeout) do
```

Comment:
Just one idea.
How about using `def chunk_id_base64` instead of struct member?
It reduces additional member cost.